websockets/cext.c

```diff
@@ -0,0 +1,128 @@
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+#include <stdlib.h> /* getenv */
+
+#ifndef HAS_SSE2
+/* setup HAS_SSE2 macro according to compiler predefined macros */
+#ifdef __SSE2__
+#define HAS_SSE2 1
+#else
+#define HAS_SSE2 0
+#endif
+#endif /* not defined HAS_SSE2 */
+
+#if HAS_SSE2
+#include <emmintrin.h>
+#endif
+
+#ifndef HAS_FAST_UNALIGNED_ACCESS
+/* This is not enabled by default */
+/* build with CFLAGS="-DHAS_FAST_UNALIGNED_ACCESS=1" to enable */
```

Comment:
I left out one of the most important piece of information.
It will just crash on some platform not supporting unaligned access if the input/output buffers are not aligned properly. So, back to the comment "there seem to be no normative information regarding alignment of PyBytes allocated buffer."